### DISCUSSION TITLE
Default Search Params before render

### DISCUSSION THREAD

**Discussion by Evild67** [TanStack/router#143] [Q&A]
What's the best way with React Location to add default search params to the url before render ?

I would like different search params on different routes

For exemple

/A?alpha=1
/B?beta=2
/D?alpha=1&beta=6

**Comment by tannerlinsley** [comment]
Use the route searchFilters option. AFK right now so more on this later.

**Reply by wscotten** [reply]
Hello!  I have been using react-location recently and ran into this issue as well.  It seems that searchFilters only works on client side navigation.  On initial page load the default search params won't be added to the url.  Is that correct?
